src/buildblock/Scanner.cxx

```diff
@@ -1751,6 +1752,23 @@ Scanner::check_consistency() const
 
   if (get_scanner_geometry() == "BlocksOnCylindrical")
     { //! Check consistency of axial and transaxial spacing for block geometry
+      if (get_num_axial_buckets() != 1)
+        {
+          warning(boost::format("BlocksOnCylindrical num_axial_buckets (%d) is greater than 1. This is not supported yet. "
+                                "Consider multiplying the number of axial_blocks_per_bucket by %d.")
+                  % get_num_axial_buckets() % get_num_axial_buckets());
+          return Succeeded::no;
+        }
+      { // Assert that each block contains an equal number of axial crystals
+        if (get_num_rings() % get_num_axial_crystals_per_bucket() != 0)
+          {
+            error(boost::format("Error in GeometryBlocksOnCylindrical: number of rings (%d) is not a multiple of the "
```

Comment:
Looks like all the other are warnings with a return, while this calls error. I think it's best to stick to warnings here, mainly such that `test_Scanner` can proceed with other scanners (otherwise we'd need `try/catch` statements there).